src/layout/ResizableSplitView.js

```diff
@@ -0,0 +1,313 @@
+  }
+  throw new Error(`Unknown resize bar state ${state}`);
+}
+
+class ResizeBar extends View {
+  _intrinsicContentSize: Size = {
+    width: 0,
+    height: 6,
+  };
+
+  _interactionState: ResizeBarState = 'normal';
+
+  constructor(surface: Surface, frame: Rect) {
+    super(surface, frame, layeredLayout);
+    this.addSubview(new ColorView(surface, frame, ''));
+    this._updateColor();
+  }
+
+  desiredSize() {
+    return this._intrinsicContentSize;
+  }
+
+  _getColorView(): ColorView {
+    return (this.subviews[0]: any);
+  }
+
+  _updateColor() {
+    this._getColorView().setColor(getColorForBarState(this._interactionState));
+  }
+
+  _setInteractionState(state: ResizeBarState) {
+    if (this._interactionState === state) {
+      return;
+    }
+    this._interactionState = state;
+    this._updateColor();
+  }
+
+  _handleVerticalPanStart(interaction: VerticalPanStartInteraction) {
+    const cursorInView = rectContainsPoint(
+      interaction.payload.location,
+      this.frame,
+    );
+    if (cursorInView) {
+      this._setInteractionState('dragging');
+    }
+  }
+
+  _handleVerticalPanMove(interaction: VerticalPanMoveInteraction) {
+    const cursorInView = rectContainsPoint(
+      interaction.payload.location,
+      this.frame,
+    );
+    if (this._interactionState === 'dragging') {
+      return;
+    }
+    this._setInteractionState(cursorInView ? 'hovered' : 'normal');
+  }
+
+  _handleVerticalPanEnd(interaction: VerticalPanEndInteraction) {
+    const cursorInView = rectContainsPoint(
+      interaction.payload.location,
+      this.frame,
+    );
+    if (this._interactionState === 'dragging') {
+      this._setInteractionState(cursorInView ? 'hovered' : 'normal');
+    }
+  }
+
+  handleInteraction(interaction: Interaction) {
+    switch (interaction.type) {
+      case 'vertical-pan-start':
+        this._handleVerticalPanStart(interaction);
+        return;
+      case 'vertical-pan-move':
+        this._handleVerticalPanMove(interaction);
+        return;
+      case 'vertical-pan-end':
+        this._handleVerticalPanEnd(interaction);
+        return;
+    }
+  }
+}
+
+export class ResizableSplitView extends View {
+  _resizingState: ResizingState | null = null;
+  _layoutState: LayoutState;
+
+  constructor(
+    surface: Surface,
+    frame: Rect,
+    topSubview: View,
+    bottomSubview: View,
+  ) {
+    super(surface, frame, noopLayout);
+
+    this.addSubview(topSubview);
+    this.addSubview(new ResizeBar(surface, frame));
+    this.addSubview(bottomSubview);
+
+    const topSubviewDesiredSize = topSubview.desiredSize();
+    this._layoutState = {
+      barOffsetY: topSubviewDesiredSize ? topSubviewDesiredSize.height : 0,
+    };
+  }
+
+  _getTopSubview(): View {
+    return this.subviews[0];
+  }
+
+  _getResizeBar(): View {
+    return this.subviews[1];
+  }
+
+  _getBottomSubview(): View {
+    return this.subviews[2];
+  }
+
+  _getResizeBarDesiredSize(): Size {
+    return nullthrows(
```

Comment:
Is it really necessary to introduce a dependency for a simple helper function? I think we can just write a function to accommodate for it here.

Reply:
We can definitely declare it ourselves, but since this `nullthrows` package is used in the React monorepo as well I figured it's okay to just add it as a dependency.

Reply:
Yeah, that makes sense.